x-conduit/src/X/Control/Monad/Trans/Resource.hs

```diff
@@ -0,0 +1,23 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module X.Control.Monad.Trans.Resource where
+
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.Resource (MonadResource, liftResourceT)
+import Control.Monad.Trans.Either (EitherT)
+
+-- | This instance is useful when adding errors to the monad used for some conduit processing
+-- when resources, like files, are involved. For example:
+--
+-- foo :: MonadResource m => ConduitM () Void (EitherT Text m) [ByteString]
+-- foo = CB.sourceFile "src/Main.hs"
+--  =$= CB.lines
+--  -- filter some lines
+--  =$= CL.map parse
+--  -- collect errors in EitherT Text m
+--  =$= CL.mapM hoistEither
+--  =$= CL.consume
+--
+-- parse :: ByteString -> Either Text ByteString
+-- parse _ = Left ":("
+instance MonadResource m => MonadResource (EitherT e m) where
```

Comment:
I am strongly against this. Why do we need an orphan here? We should be able to either call the implementation directly, or if there is some conduit API that requires MonadResource, just uplift it into a specific incarnation and don't use the generalized one for this case. In the background could send an email to Michael and Ed to see if one them would be able/willing to add a dependency to get the instance.